bin/recog_verify

```diff
@@ -48,15 +55,45 @@ if ARGV.empty?
   exit(1)
 end
 
+def validate(xmlfile_path, xsd_path)
+  schema = Nokogiri::XML::Schema(File.read(xsd_path))
```

Comment:
Nope, i'll update this

Reply:
Question: This also means that we're loading the XML documents twice every time -- once to validate using the schema, and another time to validate the examples.

I briefly looked at this. The way I'd approach solving this problem would be to make a DB factory (this is actually a good place to use one) which has two methods: One that creates a new DB from a file, and one that creates a new DB from a pre-parsed XML document (output of Nokogiri::XML(xxx)). I stopped myself from making that change since it is more invasive, and not sure if we'd want to push that change upstream to recog-go and recog-java.